app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,51 @@
+package de.xikolo.storages
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.storages.base.BaseStorage
+
+typealias RecentCourse = Pair<String, String>
+
+const val MAX_SHORTCUTS = 4
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    companion object {
+        private const val PREF_RECENT_COURSES = "preference_recent_courses"
+
+        fun addCourseToRecentCourses(
+            courseId: String,
+            title: String,
+            courses: LinkedHashSet<RecentCourse>
+        ): LinkedHashSet<RecentCourse> {
+            return courses
+                // filter duplicates
+                .filter { it.first != courseId }
+                .toMutableSet()
+                // add visited course
+                .apply { this.add(Pair(courseId, title)) }
+                // take last MAX_SHORTCUT courses
+                .reversed()
+                .take(MAX_SHORTCUTS)
+                .reversed()
```

Comment:
The second reverse does not make any sense because the collection is converted to a `Set` afterwards.

Reply:
But which implementation of `Set`? Ideally, we want to keep a `LinkedHashSet` all the time.

Reply:
It is only defined that `The returned set preserves the element iteration order of the original collection`.

Reply:
Sounds fine, or? The element iteration order should be the order of the `LinkedHashSet`. And the tests check the order as well, so it seems to work as expected?